From 3d37e3548e7541789ec500bb1beeb2505402a43c Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 27 Jan 2023 17:28:52 +0000
Subject: [PATCH] Fix build with nodejs-18

:Release Notes:
Fixes:
http://gecko.lge.com/Errors/Details/527977

:Detailed Notes:
../src/node_webos.cpp:72:120: error: no matching function for call to 'v8::ScriptOrigin::ScriptOrigin(v8::Local<v8::String>)'
   72 |         ScriptOrigin *scriptOrigin = new ScriptOrigin(String::NewFromUtf8(isolate, pathToScriptSource).ToLocalChecked());
      |                                                                                                                        ^
In file included from nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-script.h:17,
                 from nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-debug.h:10,
                 from nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8.h:29:
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:64:13: note: candidate: 'v8::ScriptOrigin::ScriptOrigin(v8::Isolate*, v8::Local<v8::Value>, int, int, bool, int, v8::Local<v8::Value>, bool, bool, bool, v8::Local<v8::Data>)'
   64 |   V8_INLINE ScriptOrigin(Isolate* isolate, Local<Value> resource_name,
      |             ^~~~~~~~~~~~
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:64:13: note:   candidate expects 11 arguments, 1 provided
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note: candidate: 'constexpr v8::ScriptOrigin::ScriptOrigin(const v8::ScriptOrigin&)'
   62 | class V8_EXPORT ScriptOrigin {
      |                 ^~~~~~~~~~~~
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note:   no known conversion for argument 1 from 'v8::Local<v8::String>' to 'const v8::ScriptOrigin&'
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note: candidate: 'constexpr v8::ScriptOrigin::ScriptOrigin(v8::ScriptOrigin&&)'
nodejs-module-webos-dynaload/3.0.2-3-r11/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note:   no known conversion for argument 1 from 'v8::Local<v8::String>' to 'v8::ScriptOrigin&&'

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/nodejs-module-webos-dynaload/+/344287 Fix build with nodejs-18]
---
 src/node_webos.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/node_webos.cpp b/src/node_webos.cpp
index 40da468..6871d04 100644
--- a/src/node_webos.cpp
+++ b/src/node_webos.cpp
@@ -69,7 +69,8 @@ Local<Value> IncludeScript(char const * pathToScriptSource, bool& exceptionOccur
 	EscapableHandleScope scope(isolate);
 	Local<Value> returnValue = Undefined(isolate);
 	Local<String> scriptSource = createV8StringFromFile(pathToScriptSource);
-	ScriptOrigin *scriptOrigin = new ScriptOrigin(String::NewFromUtf8(isolate, pathToScriptSource).ToLocalChecked());
+	ScriptOrigin *scriptOrigin = new ScriptOrigin(isolate, String::NewFromUtf8(isolate, pathToScriptSource).ToLocalChecked());
+
 	Local<Script> compiledScript(Script::Compile(currentContext, scriptSource, scriptOrigin).ToLocalChecked());
 	if(compiledScript.IsEmpty()) {
 		return returnValue;
